### PR TITLE
JSONObject nullability cleanup

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
I believe it must be squashed.

Please note that on the long term, this will be deleted as soon as all of its calls are done in kotlin, and so we don't have to deal with runtime/non-runtine exception

### CODE REVIEW

**Comment by mikehardy** [comment]
This one appears to have taken a conflict and it had lots of testing issues. Could you see what happened?

**Comment by david-allison** [comment]
Blocked on https://github.com/ankidroid/Anki-Android/pull/12141?

File: AnkiDroid/src/test/java/com/ichi2/utils/JSONObjectTest.kt [hunk 1]
```diff
@@ -661,7 +661,9 @@ class JSONObjectTest {
         val contents: MutableMap<Any?, Any?> = HashMap()
         contents[5] = 5
         try {
-            JSONObject(contents)
+            // Cast is unvalid. Used only for testing with invalid map input
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/JSONObjectTest.kt:663]: `unvalid` is not in my dictionary

**Review by david-allison — COMMENTED** [review]
One minor comment, happy to merge once the dependency is merged

**Comment by Arthur-Milchior** [comment]
Not blocked anymore. But thanks, I forgot to note it here

**Comment by mikehardy** [comment]
Ugh, conflicts

**Comment by Arthur-Milchior** [comment]
> Ugh, conflicts

With myself, as usual.

**Review by david-allison — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/utils/JSONObject.kt [hunk 2]
```diff
@@ -307,22 +306,21 @@ open class JSONObject : org.json.JSONObject, Iterable<String?> {
 
     /** deep clone this into clone.
      *
-     * Given a subtype [T] of JSONObject, and a JSONObject [j], we could do
+     * Given a subtype [T] of JSONObject, and a JSONObject [clone], we could do
      * ```
      * T t = new T();
      * j.deepClonedInto(t);
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/utils/JSONObject.kt:312]:
```suggestion
     * clone.deepClonedInto(t);
```

just a comment-code fix. Fine with merge after this fix

**Review by mikehardy — APPROVED** [review]
Only thing I can see is some example code in a comment that needs a matching rename
Approve + squash merge contingent on that

File: AnkiDroid/src/main/java/com/ichi2/utils/JSONObject.kt (hunk 2 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/utils/JSONObject.kt:312]: I just fixed this locally so it won't take conflicts again. going to push force-with-lease and merge it after it passes CI tasks locally on a rebase from main

**Comment by mikehardy** [comment]
Also took a nullability change in Models from the first commit, pushed as part of the commit, it doesn't change anything about *what* was happening here just added a nullability change similar to one that already existed

**Pull request merged**